File: Java/Q1920-Build-Array-from-Permutation.java
class Solution {
    public int[] buildArray(int[] nums) {
        if(nums.length==0){
            return nums;
        }
        int[] res=new int[nums.length];
        
        for(int i=0;i<nums.length;i++){
            res[i]=nums[nums[i]];
        }
        return res;
    }
}

File: Typescript/Q1920-Build-Array-from-Permutation.ts
function buildArray(nums: number[]): number[] {
    if(nums.length==0){
                return nums;
            }
            let res=[];
            
            for(let i=0;i<nums.length;i++){
                res.push(nums[nums[i]]);
            }
            return res;
    };